### PR TITLE
Handle single character images

### PR DESCRIPTION
**Pull request by rhatdan**
Currently you can only specify multiple character for image names
when executing podman-remote commit

podman-remote commit a b
Will complete, but will save the image without a name.

podman-remote commit a bb
Works.

This PR fixes and now returns an error if the user doees not specify an
image name to commit to.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
fixes: https://github.com/containers/podman/issues/7114
@edsantiago PTAL

**Comment by mheon** [comment]
Can you add a test?

**Comment by giuseppe** [comment]
LGTM

File: pkg/api/handlers/libpod/images.go [hunk 1]
```diff
@@ -594,11 +591,11 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// I know mitr hates this ... but doing for now
-	if len(query.Repo) > 1 {
-		destImage = fmt.Sprintf("%s:%s", query.Repo, tag)
+	if len(query.Repo) == 0 {
```
Comment (TomSweeneyRedHat) [inline pkg/api/handlers/libpod/images.go:594]: I'm always leery about " equals some number"  could you convert to `< 1`?

**Comment by edsantiago** [comment]
@rhatdan CI failures look real, in API tests:
```
not ok 154 [20-containers] POST libpod/commit?container=myfoo [] : status
#  expected: 200
#    actual: 500
```
I have to guess that your change somehow broke a corner case in APIv2.

**Comment by edsantiago** [comment]
Yep, looks like 'podman commit containername`, with no imagename, is perfectly valid:
```console
$ ./bin/podman commit --help
   podman commit [flags] CONTAINER [IMAGE]
./bin/podman run --name a alpine true                                                                                                         pr/7158
$ ./bin/podman commit a
Getting image source signatures
Copying blob 50644c29ef5a skipped: already exists
Copying blob e624f929ee9f done
Copying config d8a029cf84 done
Writing manifest to image destination
Storing signatures
d8a029cf8407bc6406697214a1bbdfe8729a845e99323f5bfd590e6ea73c974b
```
...but with this PR, that usage is no longer being allowed.

**Comment by edsantiago** [comment]
@cevich we're seeing a lot of these, on many different PRs and over a long period of time (all last week):
```
Uploading 3 artifacts for /var/tmp/go/src/github.com/containers/podman/*.log.html
Uploaded /var/tmp/go/src/github.com/containers/podman/apiv2_test.log.html
Failed to upload artifact file /var/tmp/go/src/github.com/containers/podman/integration_test.log.html: EOF
```
Does it look familiar? Is there anything you can think of that can make this more robust?

**Comment by cevich** [comment]
> Does it look familiar? Is there anything you can think of that can make this more robust?

Yes, I've heard of this, and from multiple containers repositories (i.e. it's not just a problem in one google-project or github project or cirrus-ci configuration).  Since yours represents "even more repeated complaints" now, I best bring it to the attention of CIrrus support.  That's not a guarantee that can/will do anything about it, but maybe they can help find where the problem comes from (so we can avoid it somehow).

**Comment by edsantiago** [comment]
@cevich here are three just from this PR, all demonstrating the same three-line failure: 

https://cirrus-ci.com/task/5071485364600832
https://cirrus-ci.com/task/5339766201778176
https://cirrus-ci.com/task/5621241178488832

Here's one with a zero-line failure:

https://cirrus-ci.com/task/4508535411179520

**Comment by cevich** [comment]
@edsantiago thanks, I'll get those right over to them.
